velox/functions/prestosql/tests/utils/FunctionBaseTest.h

```diff
@@ -250,6 +250,114 @@ class FunctionBaseTest : public testing::Test,
   static std::unordered_set<std::string> getSignatureStrings(
       const std::string& functionName);
 
+  /// Given an expression, a list of inputs and expected results, generate
+  /// dictionary-encoded and constant-encoded vectors, evaluate the expression
+  /// and verify the results.
+  template <typename TInput, typename TOutput>
```

Comment:
@rui-mo Rui, thank you for iterating. This method doesn't need to be a template and its implementation can be moved to .cpp. See "Review comments" commit below.

https://github.com/facebookincubator/velox/compare/main...mbasmanova:velox-1:make_decimal?expand=1

Reply:
Really appreciate your time for working on this. This commit is included.